File: packages/cli/test/fixtures/multilang/Sample.java
import java.util.HashMap;
import java.util.Optional;

public class Sample {
    private HashMap<String, User> users = new HashMap<>();

    public User createUser(String name) {
        User user = new User(generateId(), name);
        users.put(user.getId(), user);
        return user;
    }

    public Optional<User> findById(String id) {
        return Optional.ofNullable(users.get(id));
    }

    private String generateId() {
        return java.util.UUID.randomUUID().toString();
    }
}

class User {
    private String id;
    private String name;

    public User(String id, String name) {
        this.id = id;
        this.name = name;
    }

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }
}

enum UserRole {
    ADMIN,
    MEMBER,
    GUEST
}

interface Repository {
    User find(String id);
    void save(User user);
}

File: packages/cli/test/fixtures/multilang/sample.php
<?php

namespace App\Models;

use App\Contracts\Repository;

class UserService
{
    private array $users = [];

    public function create(string $name): array
    {
        $id = $this->generateId();
        $user = ['id' => $id, 'name' => $name];
        $this->users[$id] = $user;
        return $user;
    }

    public function findById(string $id): ?array
    {
        return $this->users[$id] ?? null;
    }

    private function generateId(): string
    {
        return uniqid();
    }
}

interface Repository
{
    public function find(string $id): mixed;
    public function save(array $data): void;
}

enum UserRole: string
{
    case Admin = 'admin';
    case Member = 'member';
    case Guest = 'guest';
}

function validate_email(string $email): bool
{
    return str_contains($email, '@');
}
